packages/tuono-router/src/components/RouteMatch.tsx

```diff
@@ -24,10 +24,16 @@ export const RouteMatch = ({
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const routes = useMemo(() => loadParentComponents(route), [route.id])
 
+  const routeData = isLoading ? null : data
```

Comment:
This change impacts `TuonoProps` typings:

https://github.com/tuono-labs/tuono/blob/bfd0041c6a1a14a427543f32ca80c632783004b4/packages/tuono/src/types.ts#L50-L53

The interface should become a union type to provide to typescript the correct anatomy of the runtime value, which allows a better static code analysis:

```ts
export type TuonoRouteProps<T> = { 
   data: null
   isLoading: true
 } | { 
   data: T
   isLoading: false
 } 
```

This change could be done together with:
- https://github.com/tuono-labs/tuono/issues/450

I can take care of this when I have time

Reply:
Yeah, that would be great! I can create the union type with this PR and then you address the name separately. WDYT?

Reply:
This is a breaking change since it will change the anatomy of `TuonoProps` completely.
Because of this I think this should be addressed in a different PR and have a different changeling entry.

Reply:
So you think we can merge this and then you address the other one?

Reply:
yes

Reply:
https://github.com/tuono-labs/tuono/issues/450 has been updated with new type refinements